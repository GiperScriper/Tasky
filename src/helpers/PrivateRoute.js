import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import authService from 'helpers/authService';

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    render={props =>
      authService.isAuthenticated() ? (
        <Component {...props} />
      ) : (
        <Redirect
          to={{
            pathname: '/login',
          }}
        />
      )
    }
  />
);

export default PrivateRoute;
