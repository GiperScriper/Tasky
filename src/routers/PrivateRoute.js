import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import authService from 'helpers/authService';

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route
    {...rest}
    component={props =>
      authService.isAuthenticated() ? (
        <div>
          <h1>header</h1>
          <Component {...props} />
        </div>
      ) : (
        <Redirect to="/login" />
      )
    }
  />
);

export default PrivateRoute;
