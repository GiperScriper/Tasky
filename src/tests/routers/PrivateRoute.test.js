import React from 'react';
import { shallow } from 'enzyme';
import PrivateRoute from '@/routers/PrivateRoute';

describe('<PrivateRoute />', () => {
  it('should render PrivateRoute component correctly', () => {
    const wrapper = shallow(<PrivateRoute />);
    expect(wrapper).toMatchSnapshot();
  });
});
