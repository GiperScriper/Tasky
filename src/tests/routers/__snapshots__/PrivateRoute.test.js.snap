// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> should render PrivateRoute component correctly 1`] = `
<Route
  component={[Function]}
/>
`;
